js/stdlib/ts/CBR.ts

```diff
@@ -142,6 +144,27 @@ export const BV_Address = (val: string): CBR_Address => {
   return BT_Address.canonicalize(val);
 }
 
+export const BT_Contract: BackendTy<CBR_Contract> = ({
+  name: 'Contract',
+  canonicalize: (val: unknown): CBR_Contract => {
```

Comment:
I think that this canonicalize either needs to reuse existing code (like my comment above) or it needs to be custom per-connector and do the same (i.e. this file will not expose `BT_Contract`, but will allow the connectors to define it directly, potentially using a `make_T_Contract` function. I think it would actually be the simplest to simply make `ALGO` define `T_Contract` literally as `T_UInt` and `ETH` define it as `T_Address`. In other words, there's a new NAME but not a new VALUE.

Reply:
You might think we can't do this because "We need CBR_Contract" to be one thing for all the networks, but here's a dirty secret: the CBRs aren't actually really enforced or used anywhere, because TypeScript isn't expressive enough to track it for us.